vm/rust/src/jsonrpc.rs

```diff
@@ -66,9 +74,12 @@ pub fn new_transaction_trace(tx: StarknetApiTransaction, info: BlockifierTxInfo)
                     Some(v) => Some(v.into()),
                     None => None,
                 },
-                execute_invocation: match info.execute_call_info {
-                    Some(v) => Some(v.into()),
-                    None => None,
+                execute_invocation: match info.revert_error {
+                    Some(str) => ExecuteInvocation::RevertReason(str),
+                    None => match info.execute_call_info {
+                        Some(v) => ExecuteInvocation::ExecuteInvocation(Some(v.into())),
+                        None => ExecuteInvocation::ExecuteInvocation(None),
```

Comment:
dont we need still need         `#[serde(skip_serializing_if = "Option::is_none")]`  and the change below?

```suggestion
                        None => None,
```

Reply:
I trued `#[serde(skip_serializing_if = "Option::is_none")]` with current enum and it didn't work. `None => None` just alone won't work either because `execute_invocation` is enum, but maybe I should make it `Option<ExecuteInvocation>` and get rid of Option in ExecuteInvocation::ExecuteInvocation. wdyt ? also will come up with better naming xD

Reply:
> Option<ExecuteInvocation> and get rid of Option in ExecuteInvocation::ExecuteInvocation. wdyt ?

Yeah, I think that makes the most sense.